cmd/processor/app/processor.go

```diff
@@ -649,3 +654,28 @@ func (p *Processor) setWorkersStatus(triggerInstance trigger.Trigger, status sta
 
 	return nil
 }
+
+// handleSignals creates a signal handler, so on signal processor gracefully terminated
+func (p *Processor) handleSignals() {
+	var captureSignal = make(chan os.Signal, 1)
+	signal.Notify(captureSignal, syscall.SIGTERM, syscall.SIGABRT, syscall.SIGINT)
+	p.terminateAllTriggers(<-captureSignal)
+	p.Stop()
+}
+
+func (p *Processor) terminateAllTriggers(signal os.Signal) {
+	p.logger.WarnWith("Got system signal", "signal", signal.String())
+
+	wg := &sync.WaitGroup{}
+
+	for _, triggerInstance := range p.triggers {
+		wg.Add(1)
+		// goroutine to drain trigger
```

Comment:
As @liranbg said - comment the **why** and not the **what**.
here the why is to parallelize the draining on all triggers because we have limited time.